dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -914,4 +916,85 @@ private static boolean isVarargsInvocation(
     return TypesUtils.getArrayDepth(ElementUtils.getType(lastParamElt))
         != TypesUtils.getArrayDepth(lastArgType);
   }
+
+  /**
+   * Find the declaration of the receiver of a method call in a given method tree.
+   *
+   * <p>The receiver should appear in one of two places, either as a formal parameter to the method,
+   * or as a local variable.
```

Comment:
This code only returns a local variable at the top level of the method declaration.  I think that is worth calling out.  That limitation is probably the right thing, given that a JavaExpression doesn't contain enough information to disambiguate between two different non-overlapping variable declarations with the same name.